src/keywordPreservation.ts

```diff
@@ -173,14 +177,37 @@ export const updateAssetsByAddress = (
 // preserveKeywords is the function that ultimately gets executed during export
 // to attempt to preserve keywords (ex: ##KEYWORD##) in local configuration files
 // from getting overwritten by remote values during export.
-export const preserveKeywords = (
-  localAssets: object,
-  remoteAssets: object,
-  keywordMappings: KeywordMappings
-): object => {
+export const preserveKeywords = ({
+  localAssets,
+  remoteAssets,
+  keywordMappings,
+  auth0Handlers,
+}: {
+  localAssets: object;
+  remoteAssets: object;
+  keywordMappings: KeywordMappings;
+  auth0Handlers: Pick<APIHandler, 'id' | 'identifiers' | 'type'>[];
+}): object => {
   if (Object.keys(keywordMappings).length === 0) return remoteAssets;
 
-  const addresses = getPreservableFieldsFromAssets(localAssets, keywordMappings, '');
+  const resourceSpecificIdentifiers = auth0Handlers.reduce(
+    (acc, handler): Partial<{ [key in AssetTypes]: string }> => {
+      return {
+        ...acc,
+        [handler.type]: handler.identifiers.filter((identifiers) => {
+          return identifiers !== handler.id;
+        })[0],
+      };
+    },
+    {}
+  );
```

Comment:
For historical reasons that aren't entirely understood, an Auth0 handler's identifier set always contains the resource's ID field. This is necessarily not wanted because ID will never be the same across tenants. This process instead plucks the first non-id identifier field for a resource.

Reply:
What about the feature that allows ids to be exported in the dump and then use them to include them in the payload in order to re-create clients with the same exact IDs from one tenant to another if the tenant flag that allows this is enabled? Will this change break that?

Reply:
This code only operates in the context of keyword preservation, all other functionality remains unchanged. But to your point, the `handler.identifiers` should contain the `handler.id` to enable that functionality, but that is a definition that occurs on the resource class constructor level.